File: y2017/nim/day2.nim
import strutils
import sequtils
import sugar  # wtf

let sheet = readFile("input/day2.txt")
  .strip()
  .splitLines()
  .map(r => r.splitWhitespace().map(i => parseInt(i)))
let checksum = sheet
  .map(r => minmax(r))
  .map(mm => mm[1] - mm[0])
  .foldl(a + b)  # what the actual frick

var rowSum = 0
for row in sheet:
  for v, i in row:
    for j in row[v + 1..^1]:
      if i mod j == 0 or j mod i == 0:
        let quotient = if i mod j == 0: i div j else: j div i
        rowSum += quotient

echo "checksum: $1" % [$checksum]
echo "sum of rows: $1" % [$rowSum]

File: y2017/rust/src/bin/day2.rs
use std::fs;

fn main() {
    let sheet = fs::read_to_string("../input/day2.txt").expect("bruh");

    let mut checksum = 0;
    let mut res_sum = 0;
    for row in sheet.lines() {
        let nums: Vec<i32> = row.split_whitespace()
            .map(|n| n.parse::<i32>().unwrap_or(0))
            .collect();

        checksum += nums.iter().max().unwrap_or(&0) - nums.iter().min().unwrap_or(&0);

        'search:
        for i in 0..nums.len() {
            for j in i + 1..nums.len() {
                if nums[i] % nums[j] == 0 {
                    res_sum += nums[i] / nums[j];
                    break 'search;
                }
                if nums[j] % nums[i] == 0 {
                    res_sum += nums[j] / nums[i];
                    break 'search;
                }
            }
        }
    }

    dbg!(checksum);
    dbg!(res_sum);
}
